### DISCUSSION TITLE
Report - (3rd Sem.) Week 4 📢

### DISCUSSION THREAD

**Discussion by Nahlam4** [SE-TINF22B6/DHBWhub#108] [Weekly Report]
Welcome to the fourth report of our progress. We are excited to share the latest news with you! 🥳

### CI Workflow 🗺️ 
We have established or are still working on refining our CI/CD setup of our project. If you want to view a rather detailed description
of our setup, you can just read it [here](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/CICD_setup.md). Otherwise, the following graph depicts our current CI/CD workflow:
>![UsecaseDiagramGeneral](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/diagrams/CI.CD_workflow.png?raw=true)

#### Development Environment:
- Maven helps organizing our project, its dependencies and allows us to see, whether our changes may conflict our local branch through building processes
- Utility tools like JUnit (Java) or React Testing Library (React, JavaScript) are used for unit testing our components in an isolated environment and ensure a proper code quality
- SonarLint is a plugin which you can download through your IDE. It analyzes your code locally after code smells, bugs and other errors and proposes best practices. Other team members can also see the analysis when connected to a SonarQube/-Cloud server.

#### Source Code Repository Environment:
After every push or pull request on our main branches, following workflows are triggered by Github Actions.
- CodeQL: GitHub tool, which analyzes our code for vulnerabilities and security issues
- Maven: Automatically runs a Maven Build and Test of our whole project.
- React: Automatically runs a React Build and Test of solely the frontend package.

One exception is the DependaBot. It runs periodically and checks whether our dependencies are outdatet or not.
 
### Frontend development 💻

✅ We began setting up our homepage according to our mockup and created the header which you can see below:

>![homepage_header](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/e5282e6c-f62c-4670-9c4d-83679c905adb)

✅ When clicking on the "login"-button, the login view will pop up:
>![login](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/59d91690-7236-41c8-811b-e25fad503254)

### Sprint 2: What have we achieved ❓  What's to come❔
- ✅Agreeing on our login-options and the way we persist account data
- ✅Using Figma for further design of the post-page
- ✅Plan Sign-in process
- ✅Implementation of ER-diagram in the database
- ✅Setup Server for hosting the database
    - ✅Setup database on the server
- ✅Building the web interface
    - ✅Creating Signup/Login-window
    - ✅Creating navbar
- 🚧Implementing First Unit Tests
(- 🚧Integrating SonarQube for code analysis)
- 🚧Discussing possible API's
- 🚧Define more use cases and user stories
- 🚧Removing Herobrine 

Until next week 💯

Your DHBWhub Team

**Comment by sfrancisw** [comment]
Hello,
Awesome progress you got there. The blog looks very detailed and informative. I like that you are linking to a more detailed explanation of your CI/CD setup to give people who want to dive deeper into the topic the opportunity to do so. The frontend is also looking nice. We're excited to see where the projec is heading and how the frondend will evolve.

Your team time2shine

**Comment by MargotGsb** [comment]
Dear DHBWhub team,
I really like your visualization of your CI/CD workflow. It is very easy to understand.
I also think, that you made great progress, in your UI. It looks very good and intuitive. The achievements you made during this sprint, seem really good in general to me. And the report of this week is really easy to read and well structured.
Keep going. We are all looking forward to see how this project turns out.

Your team Plapy
